lib/money_column/active_record_hooks.rb

```diff
@@ -74,9 +74,9 @@ def write_money_attribute(column, money)
 
     def compatible_currency?(money, options)
       currency_column = options[:currency_column]
-      currency = options[:currency] ||
-        @money_raw_new_attributes[currency_column.to_sym] ||
-        try(currency_column)
+      currency = options[:currency]
+      currency ||= @money_raw_new_attributes[currency_column.to_sym] if @money_raw_new_attributes
+      currency ||= try(currency_column)
```

Comment:
Why is `@money_raw_new_attributes` even needed? 🤔

Reply:
it's required to make https://github.com/Shopify/money/pull/409/files#diff-0fdde4bd8a1f5fb142f661be86c568d43b1061f447e188d1d585ab1974458548R416 test pass. Let me know if you have another way you're thinking of that doesn't requires this, but also keeps the test passing

Reply:
I'm cool with merging this for now and exploring other options in parallel 👍🏼